src/java.xml/share/classes/module-info.java

```diff
@@ -27,6 +27,92 @@
  * Defines the Java API for XML Processing (JAXP), the Streaming API for XML (StAX),
  * the Simple API for XML (SAX), and the W3C Document Object Model (DOM) API.
  *
+ * <h2 id="ConfigurationFile">Configuration File</h2>
+ * The java.xml (JAXP) Configuration File is used for configuring factories in the
+ * <a href="#LookupMechanism">Factory Lookup Mechanism</a>, secure processing,
+ * and Catalog API.
+ *
+ * <h3>Format</h3>
+ * The configuration file is in standard {@link java.util.Properties} format.
+ * <p>
+ * The keys are:
+ * <ul>
+ * <li>
+ * For the <a href="#LookupMechanism">Factory Lookup Mechanism</a>, System properties
+ * as listed in column System Property of the table <a href="#Factories">JAXP Factories</a>.
+ * </li>
+ * <li>
+ * <p>
+ * For secure processing, {@code Full Name} in
+ * table <a href="#Properties">Implementation Specific Properties</a> and
+ * <a href="#Features">Features</a>
+ * </li>
+ * <li>
+ * <p>
+ * For the Catalog API, {@code System Property} in table {@code Catalog Features}
+ * of class {@link javax.xml.catalog.CatalogFeatures CatalogFeatures}
+ * </li>
+ * </ul>
+ * <p>
+ * The values are:
+ * <ul>
+ * <li>
+ * For the <a href="#LookupMechanism">Factory Lookup Mechanism</a>, the fully
+ * qualified name of the implementation class.
+ * </li>
+ * <li>
+ * <p>
+ * For secure processing, {@code Value} in
+ * table <a href="#Properties">Implementation Specific Properties</a> and
+ * <a href="#Features">Features</a>
+ * </li>
+ * <li>
+ * <p>
+ * For the Catalog API, {@code value} in table {@code Catalog Features}
+ * of class {@link javax.xml.catalog.CatalogFeatures CatalogFeatures}
+ * </li>
+ * </ul>
+ * <p>
+ * Below are examples on what can be placed in the configuration file:
+ * <pre>
+ *     {@code javax.xml.parsers.DocumentBuilderFactory=packagename.DocumentBuilderFactoryImpl}
+ *     {@code jdk.xml.entityExpansionLimit=2000}
+ * </pre>
+ *
+ *
+ * <h3 id="CF_Default">Default</h3>
+ * The JDK may be distributed with a configuration file called {@code jaxp.properties}
+ * that is used by all factories listed in <a href="#Factories">the Factories table</a> below.
+ * It may also include one called {@code stax.properties} that will be used by
+ * the StAX factories for the <a href="#LookupMechanism">Factory Lookup Mechanism</a>
+ * only. The files, if exist, are typically located in the {@code conf}
+ * directory of the Java installation.
```

Comment:
If jdk.xml.config.file is not set, should the reader assume it's default value is ${java.home}/conf/jaxp.properties? If so, they this is an implementation specific default, right?

Reply:
Both API and Impl, will make it clearer, and add a section under the implNote section.